src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -624,11 +627,30 @@ private FormulaValue GetVariableOrFail(ResolvedObjectNode node, ISymbolSlot slot
                 var value = _symbolValues.Get(slot);
                 if (value != null)
                 {
+                    value = AdjustDateorDateTimeValue(value);

```

Comment:
>AdjustDateorDateTimeValue [](http://example.com/codeflow?start=28&length=25)

Do we need this yet - or can we defer until point of use?

There are too many ways we might get a FormulaValue

Reply:
Sure, updated.